Client/Frontend/Browser/TabScrollController.swift

```diff
@@ -282,9 +289,9 @@ extension TabScrollingController: UIScrollViewDelegate {
 
         if (decelerate || (toolbarState == .animating && !decelerate)) && checkScrollHeightIsLargeEnoughForScrolling() {
             if scrollDirection == .up {
-                showToolbars(animated: true)
+                showToolbars(animated: !isTabShowingPDF)
             } else if scrollDirection == .down {
-                hideToolbars(animated: true)
+                hideToolbars(animated: !isTabShowingPDF)
```

Comment:
Although not mentioned in the bug description, this animation has the same problem as show.

Reply:
On a PDF doc, if one partially hides/shows the toolbar and lifts their finger, it will snap to completion instead of animating to completion. I think it looks fine to do this, it is very unlikely to be noticed.

Reply:
Maybe drop in a comment here explaining why we don't animate if we're showing a PDF just so we don't accidentally regress on this in the future.

It would be nice if we could just have a different animation in this case that doesn't interfere with the heights. Like just fade in place. Still though, getting rid of the flicker is more important than an animation. We can always revisit this later as a visual polish thing.